sources/packages/backend/apps/queue-consumers/src/services/metrics/metrics.service.ts

```diff
@@ -0,0 +1,156 @@
+    collectDefaultMetrics({ labels: { app: DEFAULT_METRICS_APP_LABEL } });
+  }
+
+  /**
+   * Refreshes the metrics job counts for all queues by retrieving the current
+   * job counts for 'active', 'completed', 'failed', 'delayed', 'waiting'.
+   * The job counts are then used to update the gauge metric for each individual queue.
+   */
+  private async refreshJobCountsMetrics(): Promise<void> {
+    const monitoredQueues = await this.getMonitoredQueues();
+    await processInParallel(
+      (queue: MonitoredQueue) => this.refreshJobCountsMetricsForQueue(queue),
+      monitoredQueues,
+    );
+  }
+
+  /**
+   * Refreshes the metrics job counts for the given queue by retrieving
+   * the current values from Redis and updating the gauge metric.
+   * @param queue monitored queue details.
+   */
+  private async refreshJobCountsMetricsForQueue(
+    queue: MonitoredQueue,
+  ): Promise<void> {
+    const queueType = queue.queueModel.isScheduler ? "scheduler" : "consumer";
+    const queueJobCounts = await queue.provider.getJobCounts();
+    Object.keys(queueJobCounts).forEach((jobCountEvent: string) => {
+      this.jobCountsGauge.set(
+        {
+          queueName: queue.provider.name,
+          queueEvent: jobCountEvent,
+          queueType,
+        },
+        queueJobCounts[jobCountEvent],
+      );
+    });
+  }
+
+  /**
+   * Associate all queues with their respective counters to track the events.
+   * The counters are associated for every queue event.
+   */
+  async associateQueueEventsCountersMetrics(): Promise<void> {
+    const queues = await this.getMonitoredQueues();
+    queues.forEach((queue) => this.associateCountersToQueueEvents(queue));
+  }
+
+  /**
+   * Associates a counter to every queue event of the given queue.
+   * This allows the metrics to track the events of the queues.
+   * @param queue monitored queue details.
+   */
+  private associateCountersToQueueEvents(queue: MonitoredQueue): void {
+    const queueName = queue.provider.name;
+    const queueType = queue.queueModel.isScheduler ? "scheduler" : "consumer";
+    Object.values(QueuesMetricsEvents).forEach(
+      (queueEvent: QueuesMetricsEvents) => {
+        this.logger.log(
+          `Associating metric counter for queue '${queueName}' event '${queueEvent}'.`,
+        );
+        queue.provider.on(queueEvent, () => {
+          this.jobsEventsCounter.inc({
+            queueName,
+            queueEvent,
+            queueType,
+          });
+        });
+      },
+    );
+  }
+
+  /**
+   * Get the list of monitored queues.
+   * If not loaded yet, load it once.
+   * @returns list of monitored queues.
+   */
+  private async getMonitoredQueues(): Promise<MonitoredQueue[]> {
+    if (!this.monitoredQueueProviders) {
+      const queues = await this.queueService.queueConfigurationModel();
+      this.monitoredQueueProviders = queues
+        .filter((queueModel) => queueModel.isActive)
```

Comment:
Are we not ignoring the FT queues here?

Reply:
The full-time toggle is not checked here. If the queues are not executing they will not produce any event.
The events will be associated but never triggered. I do not see any harm in it. Please let me know if further discussion would be required on this.

Reply:
Not required and I am good based on what we discussed.